string-manipulation-3/src/is-anagram.js

```diff
@@ -0,0 +1,24 @@
+/* exported isAnagram */
+function isAnagram(firstString, secondString) {
+  var charCountFirst = [];
+  var charCountSecond = [];
+  var noSpace = firstString.split(' ');
+  firstString = noSpace.join('');
+  noSpace = secondString.split(' ');
+  secondString = noSpace.join('');
+  if (firstString.length !== secondString.length) {
+    return false;
+  }
+  for (var i = 0; i < firstString.length; i++) {
+    if (secondString.includes(firstString[i])) {
+      charCountFirst = firstString.split(firstString[i]);
+      charCountSecond = secondString.split(firstString[i]);
+      if (charCountFirst.length !== charCountSecond.length) {
+        return false;
+      }
+    } else {
+      return false;
+    }
+  }
+  return true;
```

Comment:
Every time you use the `split` or `join` method, that is running a loop for you. This solution is pretty intensive in terms of total operations, there are quite a few loops at play here. To try and reduce the overall number of loops, here is a recommendation a path to follow: rather than chopping up the words one letter at a time, you should total up how many of each letter there is in each word, and compare the totals for each letter. If any total doesn't match between both words, then it is not an anagram.